src/app/lib/profiles/profile.validator.ts

```diff
@@ -115,6 +115,12 @@ export class ProfileValidator implements IProfileValidator {
 		}).validate(data, { allowUnknown: true, stripUnknown: true });
 
 		if (error !== undefined) {
+			const missingData = error.details.some((d) => d.path[0] === "data" && d.type === "any.required");
+
+			if (missingData && typeof (data as any).ciphertext === "string") {
+				throw new Error("PasswordRequired");
+			}
+
```

Comment:
this feels quite obscure, any option we can check for a password prior to validating to make sure the provide password can decrypt the data?

Reply:
feels a lot cleaner now